pkg/operator/resource/resourceapply/core.go

```diff
@@ -34,6 +34,14 @@ func ApplyNamespace(client coreclientv1.NamespacesGetter, recorder events.Record
 	existingCopy := existing.DeepCopy()
 
 	resourcemerge.EnsureObjectMeta(modified, &existingCopy.ObjectMeta, required.ObjectMeta)
+
+	// allow removing runlevels since the metadata are otherwise merged
+	_, requiredRunLevelPresent := required.ObjectMeta.Labels["run-level"]
+	if _, ok := existingCopy.ObjectMeta.Labels["run-level"]; ok != requiredRunLevelPresent && !requiredRunLevelPresent {
```

Comment:
the add case works?

`ok && !requiredRunLevelPresent` is simpler

Reply:
The add case is handled by the `EnsureObjectMeta()` function above.

Good simplification, I modified the condition in latest revision.